### PR TITLE
[Fix #2232] Fix false positive in Lint/FormatParameterMismatch for argument with splat operator

### LINKED ISSUE #2232: False positive in Lint/FormatParameterMismatch for argument with splat operator
Extracted from #2123. Reported by @dreyks.

What about the opposite situation?

```ruby
v = [1, 2]
format('%d%d', *v)
```

```
W: Number arguments (1) to format mismatches expected fields (2).
```

this should probably be ignored too.

### PR DESCRIPTION
**Pull request by dreyks**

### CODE REVIEW

File: lib/rubocop/cop/lint/format_parameter_mismatch.rb [hunk 1]
```diff
@@ -57,6 +57,18 @@ def named_mode?(node)
             .scan(NAMED_FIELD_REGEX).count > 0
         end
 
+        def splat_node?(node)
+          return false if percent?(node)
+
+          _receiver_node, _method_name, *args = *node
+
+          args[1..-1].each do |arg|
```
Comment (jonas054) [inline lib/rubocop/cop/lint/format_parameter_mismatch.rb]: `args[1..-1].any? { |arg| arg.type == :splat }` is more concise.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## master (unreleased)
 
+* [#2232](https://github.com/bbatsov/rubocop/issues/2232): Fix false positive in Lint/FormatParameterMismatch for argument with splat operator. ([@dreyks][])
```
Comment (jonas054) [inline CHANGELOG.md]: Put backquotes around `Lint/FormatParameterMismatch`.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,8 @@
 
 ## master (unreleased)
```
Comment (jonas054) [inline CHANGELOG.md]: There should be a `### Bug Fixes` subheading.

**Comment by dreyks** [comment]
Fixed problems that @jonas054 pointed out.

Should I squash and force push or will it be done on merge?

File: lib/rubocop/cop/lint/format_parameter_mismatch.rb [hunk 4]
```diff
@@ -57,6 +57,14 @@ def named_mode?(node)
             .scan(NAMED_FIELD_REGEX).count > 0
         end
 
+        def splat_node?(node)
```
Comment (bbatsov) [inline lib/rubocop/cop/lint/format_parameter_mismatch.rb]: I'd name this `splat_args?` or something like this.

**Comment by bbatsov** [comment]
Apart from my remark, you'll have to squash your commits together.

File: lib/rubocop/cop/lint/format_parameter_mismatch.rb (hunk 4 above)
Comment (dreyks) [inline lib/rubocop/cop/lint/format_parameter_mismatch.rb]: node_with_splat_args? or too long?

Comment (bbatsov) [inline lib/rubocop/cop/lint/format_parameter_mismatch.rb]: Fine by me.

**Comment by dreyks** [comment]
should I rebase and force push? or how to do it properly?

**Comment by bbatsov** [comment]
Use rebase & force push.

**Comment by dreyks** [comment]
done

**Comment by bbatsov** [comment]
:+1: Thanks!

**Pull request merged**